src/main/java/org/testng/internal/TestResult.java

```diff
@@ -320,4 +336,90 @@ public boolean wasRetried() {
   public void setWasRetried(boolean wasRetried) {
     this.m_wasRetried = wasRetried;
   }
+
+  public List<ITestNGMethod> getSkipCausedBy() {
+    if (this.m_status != SKIP) {
+      return Lists.newArrayList();

```

Comment:
2 calls will return 2 different instances.

Reply:
I didn't quite understand this comment. can you please explain ? If the `TestResult` is not for a skipped method, then an empty list is returned. For everything else, the reason is computed and then returned.

Reply:
@juherr - Just trying to understand.. How would that matter? I cache a valid list, because it needs to be computed and so I was trying to save the effort there. But this is just an empty list. How would it matter to an end user ?

Reply:
You are right, it's a little optimization but by default:
> Constructs an empty list with an initial capacity of ten.
https://docs.oracle.com/javase/8/docs/api/java/util/ArrayList.html#ArrayList--

Maybe use Collections.emptyList instead